lib/data-builder.js

```diff
@@ -80,3 +75,14 @@ function fillBadRequestError(data, err) {
 function fillInternalError(data, err) {
   data.message = httpStatus[data.statusCode] || 'Unknown Error';
 }
+
+function cloneAllProperties(data, err) {
+  data.name = err.name;
+  data.message = err.message;
+  for (var p in err) {
+    if ((p in data)) continue;
+    data[p] = err[p];
+  }
+  // NOTE err.stack is not an enumerable property
```

Comment:
Please add a similar note at the top for `err.message` and `err.name`.

Also perhaps explain why we are adding stack as the last property, which is to keep the long stack traces at the bottom of the error responses. When the stack trace is in the middle, it is difficult to find the end of the stack trace where the error properties are continued.